app/views.py

```diff
@@ -1,23 +1,106 @@
 from flask import Flask, request, jsonify
 from flask_restful import Api, Resource
 import json
+import random
+
+from .models import User, Requests, Requests_Schema
 
 app = Flask(__name__)
 api = Api(app)
 
-class UserAPI(Resource):
+users = []
+requests = []
+
+class UserSignupAPI(Resource):
     def post(self):
-        pass
+        user = request.get_json()
+        username = user['username']
+        department = user['department']
+        email = user['email']
+        password = user['password']
+
+        u = User(
```

Comment:
I would suggest that you don't create a user before doing the checking (validation) of the fields.